src/wkt/src/message.rs

```diff
@@ -20,27 +20,52 @@ use crate::AnyError as Error;
 /// A trait that must be implemented by all messages.
 ///
 /// Messages sent to and received from Google Cloud services may be wrapped in
-/// [Any][crate::any::Any]. `Any` uses a `@type` field to encoding the type
+/// [Any][crate::any::Any]. `Any` uses a `@type` field to encode the type
 /// name and then validates extraction and insertion against this type.
 pub trait Message {
     /// The typename of this message.
     fn typename() -> &'static str;
 
+    /// Returns the serializer for this message type
     #[doc(hidden)]
-    /// Store the value into a JSON object.
-    fn to_map(&self) -> Result<Map, Error>
+    fn serializer() -> Box<dyn MessageSerializer<Self>>
     where
-        Self: serde::ser::Serialize + Sized,
+        Self: Sized + serde::ser::Serialize + serde::de::DeserializeOwned + 'static,
     {
-        to_json_object(self)
+        Box::new(DefaultSerializer::<Self>::new())
     }
+}
 
-    #[doc(hidden)]
-    /// Extract the value from a JSON object.
-    fn from_map(map: &Map) -> Result<Self, Error>
-    where
-        Self: serde::de::DeserializeOwned,
-    {
+/// Internal API for message serialization.
+/// This is not intended for direct use by consumers of this crate.
+#[doc(hidden)]
+pub trait MessageSerializer<T> {
```

Comment:
We do not want downstream crates to implement this, so it should be: 

```rs
pub(crate) trait MessageSerializer<T> { ... }
```

and then we will have to deal with all of the warnings from running:
```
cargo clippy -p google-cloud-wkt -- --deny warnings
```

Reply:
Done.